scripts/corelli/calibration/database.py

```diff
@@ -7,118 +7,198 @@
 
 # File that defines functionality for Corelli calibration database
 
-import pathlib
 from datetime import datetime
+import enum
+import pathlib
 import re
-from typing import Optional
+from typing import List, Optional, Tuple, Union
 
-from mantid.dataobjects import TableWorkspace
-from mantid.api import WorkspaceGroup, Workspace
+from mantid.dataobjects import EventWorkspace, TableWorkspace,  Workspace2D
+from mantid.api import mtd, Workspace, WorkspaceGroup
 from mantid.simpleapi import CreateEmptyTableWorkspace, SaveNexusProcessed, LoadNexusProcessed
 
+# Custom type aliases
+CalibrationInputSetTypes = Union[str, TableWorkspace, Workspace2D, List[TableWorkspace], List[str], WorkspaceGroup]
+InputWorkspaceTypes = Union[str, Workspace2D, EventWorkspace]
+
+
+class TableType(enum.Enum):
+    r"""Allowed types of calibration-related tables to be saved and/or loaded"""
+    CALIBRATION = 'calibration'
+    FIT = 'fit'
+    MASK = 'mask'
+
+    @classmethod
+    def assert_valid_type(cls, query_type: str):
+        r"""
+        Assert if a query type is a valid table type, one of 'calibration', 'fit', 'mask'
+        :param query_type: input query type.
+        :raises: AssertionError
+        """
+        assert query_type in [table_type.value for table_type in list(cls)], f'{query_type} is not a valid table type'
 
-def init_corelli_table( name:Optional[str] = None )->TableWorkspace :
+
+def init_corelli_table(name: Optional[str] = None, table_type='calibration') -> TableWorkspace:
     """
     Function that initializes a Corelli calibration TableWorkspace columns
     """
-    table:TableWorkspace = CreateEmptyTableWorkspace(OutputWorkspace=name) if name else CreateEmptyTableWorkspace()
+    table: TableWorkspace = CreateEmptyTableWorkspace(OutputWorkspace=name) if name else CreateEmptyTableWorkspace()
 
-    table.addColumn(type="int", name="Detector ID")
-    table.addColumn(type="vector_double", name="Detector Position")
+    # expected columns declarations (column_type, column_name) for each type of calibration table
+    column_declarations = {'calibration': [('int', 'Detector ID'), ('double', 'Detector Y Coordinate')],
+                           'mask': [('int', 'Detector ID')]}
+    for column_type, colum_name in column_declarations[table_type]:
+        table.addColumn(type=column_type, name=colum_name)
     return table
 
 
-def has_valid_columns(table:TableWorkspace)->bool:
+def has_valid_columns(table: TableWorkspace, table_type='calibration') -> bool:
     """
     Check if table column names are valid Corelli calibration tables
     """
+    names_expected = {'calibration': ['Detector ID', 'Detector Y Coordinate'], 'mask': ['Detector ID']}
```

Comment:
Do we have specific names expected for the `fit` type?

Reply:
The `fit` workspaces are not `TableWorkspace` objects but `Workspace2D`. No need to check for valid columns there

Reply:
Got it, thanks!